app/controllers/admin/users_controller.rb

```diff
@@ -0,0 +1,69 @@
+class Admin::UsersController < ApplicationController
+	# before_action :authenticate_user!
+	before_action :set_user, only: [:update, :show, :destroy, :logs]
+
+	def index
+    @filtered_result = User.filter_by(params[:page], filter_params.to_h, sort_params.to_h)
+    @object_without_pagination = @filtered_result.data.distinct('users.id')
+    @users = @object_without_pagination.page(params[:page]).per(30)
+    respond_to do |format|
+      if request.xhr?
+        format.html {render partial: 'admin/users/table', locals: {users: @users}}
+      else
+        format.html
+      end
+    end
+	end
+
+	def show
+		@user = User.find(params[:id])
```

Comment:
@anbublacky - this is not required because of the `before_action` right?

Reply:
Done